src/Pos/Block/Network/Listeners.hs

```diff
@@ -78,24 +79,23 @@ stubListenerConv' = unproxy $ \(_ :: Proxy ssc) ->
 handleGetHeaders
     :: forall ssc m.
        (WorkMode ssc m)
-    => (ListenerSpec m, OutSpecs)
-handleGetHeaders = listenerConv $ \_ peerId conv -> do
-    logDebug $ "handleGetHeaders: request from " <> show peerId
-    handleHeadersCommunication conv
+    => m (ListenerSpec m, OutSpecs)
+handleGetHeaders = reifyMsgLimit (Proxy @MsgGetHeaders) $ \limitProxy ->
+    return $ listenerConv $ \_ peerId conv -> do
+        logDebug $ "handleGetHeaders: request from " <> show peerId
+        handleHeadersCommunication conv limitProxy
 
 handleGetBlocks
     :: forall ssc m.
        (WorkMode ssc m)
-    => (ListenerSpec m, OutSpecs)
-handleGetBlocks =
-    -- NB #put_checkBlockSize: We can use anything as maxBlockSize
-    -- here because 'put' doesn't check block size.
-    reify (0 :: Byte) $ \(_ :: Proxy s0) ->
-    listenerConv $
+    => m (ListenerSpec m, OutSpecs)
+handleGetBlocks = reifyMsgLimit (Proxy :: Proxy MsgGetBlocks) $
+    \(_ :: Proxy s) ->
+      return $ listenerConv $
         \_ __peerId (conv :: ConversationActions
-                               (MsgBlock s0 ssc)
-                               MsgGetBlocks m) ->
-        whenJustM (recv conv) $ \mgb@MsgGetBlocks{..} -> do
+                               (MsgBlock ssc)
+                               (LimitedLength s MsgGetBlocks) m) ->
```

Comment:
What for? I don't think we would like to change MsgGetBlocks max size eventually